chainstate/tx-verifier/src/transaction_verifier/transferred_amount_check.rs

```diff
@@ -348,6 +368,35 @@ pub fn check_transferred_amount_in_reward<U: UtxosView, P: PoSAccountingView>(
     Ok(())
 }
 
+pub fn calculate_tokens_burned_in_outputs(
+    outputs: &[TxOutput],
+) -> Result<BTreeMap<CoinOrTokenId, Amount>, ConnectTransactionError> {
```

Comment:
If this is only for tokens, why are we returning CoinOrTokenId? 

Do we need this kind of fracturing, where we treat half for coins and half for tokens?